pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -319,10 +319,6 @@ func (p ParamSpec) ValidateType() *apis.FieldError {
 // definition of `properties` section and the type of a PropertySpec is allowed.
 // (Currently, only string is allowed)
 func (p ParamSpec) ValidateObjectType() *apis.FieldError {
-	if p.Type == ParamTypeObject && p.Properties == nil {
-		return apis.ErrMissingField(fmt.Sprintf("%s.properties", p.Name))
-	}
-
```

Comment:
@chuangw6, based on our discussion,  I updated this line to:
```
if p.Type == ParamTypeObject && p.Default != nil && p.Properties == nil {
```
This will enforce that if the defaults are specified then properties must also be provided.